src/app/panel/auth/SignUp.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import { TextInput, Button } from '@tremor/react';
+import { APIService } from '@/services/api';
+
+const SignUp: React.FC = () => {
+  const [username, setUsername] = useState<string>('');
+  const [email, setEmail] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [confirmPassword, setConfirmPassword] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string | null>('');
+  const [isFormValid, setIsFormValid] = useState<boolean>(false);
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    APIService.signUp(username, email, password);
+  };
+
+  const onUsernameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.target.value = e.target.value.replace(/[^a-zA-Z0-9]/g, '');
+    setUsername(e.target.value);
+    validateForm();
+  }
+
+  const onEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.target.value = e.target.value.replace(/[^a-zA-Z0-9]/g, '');
```

Comment:
This is missing an '@' and a '.' Rather crucial in an email I think.

```suggestion
  const onEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    e.target.value = e.target.value.replace(/[^a-zA-Z0-9@.]/g, '');
```

Reply:
I totally forgot to go back to this part, let me check again <3